cqf-fhir-cr/src/main/java/org/opencds/cqf/fhir/cr/measure/r4/R4MultiMeasureService.java

```diff
@@ -0,0 +1,147 @@
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.StringUtils;
+import org.hl7.fhir.instance.model.api.IIdType;
+import org.hl7.fhir.r4.model.Bundle;
+import org.hl7.fhir.r4.model.Bundle.BundleType;
+import org.hl7.fhir.r4.model.CanonicalType;
+import org.hl7.fhir.r4.model.Endpoint;
+import org.hl7.fhir.r4.model.IdType;
+import org.hl7.fhir.r4.model.Measure;
+import org.hl7.fhir.r4.model.MeasureReport;
+import org.hl7.fhir.r4.model.Parameters;
+import org.hl7.fhir.r4.model.Resource;
+import org.opencds.cqf.fhir.api.Repository;
+import org.opencds.cqf.fhir.cr.measure.CareGapsProperties;
+import org.opencds.cqf.fhir.cr.measure.MeasureEvaluationOptions;
+import org.opencds.cqf.fhir.cr.measure.common.MeasureEvalType;
+import org.opencds.cqf.fhir.cr.measure.r4.utils.R4MeasureServiceUtils;
+import org.opencds.cqf.fhir.utility.Ids;
+import org.opencds.cqf.fhir.utility.builder.BundleBuilder;
+import org.opencds.cqf.fhir.utility.monad.Either3;
+import org.opencds.cqf.fhir.utility.repository.Repositories;
+
+// Alternate MeasureService call to Process MeasureEvaluation for the selected population of subjects against n-number
+// of measure resources. The output of this operation would be a bundle of MeasureReports instead of MeasureReport.
+
+public class R4MultiMeasureService {
+    private static final org.slf4j.Logger log = org.slf4j.LoggerFactory.getLogger(R4MultiMeasureService.class);
+    private final Repository repository;
+    private final MeasureEvaluationOptions measureEvaluationOptions;
+    private final CareGapsProperties careGapsProperties;
+
+    public R4MultiMeasureService(
+            Repository repository,
+            MeasureEvaluationOptions measureEvaluationOptions,
+            CareGapsProperties careGapsProperties) {
+        this.repository = repository;
+        this.measureEvaluationOptions = measureEvaluationOptions;
+        this.careGapsProperties = careGapsProperties;
+    }
+
+    public Bundle evaluate(
+            List<Either3<CanonicalType, IdType, Measure>> measures,
+            String periodStart,
+            String periodEnd,
+            String reportType,
+            String subjectId,
+            Endpoint contentEndpoint,
+            Endpoint terminologyEndpoint,
+            Endpoint dataEndpoint,
+            Bundle additionalData,
+            Parameters parameters,
+            String productLine,
+            String practitioner) {
+
+        var repo = Repositories.proxy(repository, true, dataEndpoint, contentEndpoint, terminologyEndpoint);
+
+        var subjectProvider = new R4RepositorySubjectProvider();
+
+        var processor = new R4MeasureProcessor(repo, this.measureEvaluationOptions, subjectProvider);
+
+        R4MeasureServiceUtils r4MeasureServiceUtils = new R4MeasureServiceUtils(repository);
+        try {
+            r4MeasureServiceUtils.ensureSupplementalDataElementSearchParameter();
+        } catch (NotImplementedOperationException e) {
+            log.warn(
```

Comment:
issue: I don't think the exception here can be hit. There's a catch inside the util class. I'd suggest just removing this exception block.